aip/general/0193.md

```diff
@@ -62,11 +62,90 @@ a [`PreconditionFailure`][PreconditionFailure].
-required, the service **should** use [`google.rpc.LocalizedMessage`][details]
-in the `details` field.
+The [`google.rpc.Status`][Status] error payload **should** contain two
+error messages.
+
+1. The payload **must** contain an error message in the `message` field
+   found at the root of [`Status`][Status].
+1. The payload **should** contain
+   [`google.rpc.LocalizedMessage`][LocalizedMessage] in `details` on
+   `Status`. This is the location of the second error message.
+
+The error message on `Status` should,
+
+- Be developer-facing, human-readable, and in English
+- Both expalin the error and offer an actionable resolution to it
+  ([citation](https://cloud.google.com/apis/design/errors#error_model))
+
+Once launched the value of `message` at the root-level of `Status`
+**must not** change for the error payload to be backwards compatible.
+
+When including `LocalizedMessage`, both the `locale` and `message`
+fields **must** be present.
+
+#### Changing error messages
+
+A new, additional error message **may** be added because the error
+message at the root of the payload **must not** change once launched.
+This second error messages allows for continuous, iterative
+improvement. It is potentially displayed directly to end users by
+client application.
+
+When adding a new error messsage with `LocalizedMessage`, `ErrorInfo`
+**must** be introduced before or concurrently. If there are dyanmic
+variables found in the message, these variables must be included in the
+`metadata` field on `ErrorInfo`, a `map<string, string>`. See, "[Dynamic
+variables](#dynamic-variables)".
+
+**Warning:** If `LocalizedMessage` is released without `ErrorInfo` even
+the new error message **must not** change. To do so, would be backwards
+incompatible, effectively locking all error messages in the payload.
+
+If the service is location aware, the value of `locale` **should**
+change.  See, "[Localization](#localization)". Otherwise, the field
+**may** contain the service's default locale, e.g.  "en-US", as its
+value, always, allowing the opportunity to add localization later.
+
+#### Dynamic variables
+
+The best, actionable error messages include dynamic segments. These
+variable parts of the message are specific to a particular request.
+Consider the following example:
+
+> The Book, "The Great Gatsby", is unvailable at the Library, "Garfield
+> East". It is expected to be available again on 2199-05-13.
+
+**Note:** The preceeding error message is made actionable by the
+context, both that orignates from the request, the title of the Book and
+the name of the Library, and by the information that is known only by
+the service, i.e. the expected return date of the Book.
+
+All dynamic variables found in error messages **must** also appear in
+the `map<string, string>` field, `metadata` (found on the *required*
+`ErrorInfo`).  For example, the `metadata` map for the sample error
+message above **must** include *at least* the following key/value pairs:
```

Comment:
I'm not sure that an example should contain a requirement word. Perhaps replace it with `will`